### ISSUE TITLE
analyze: rewriter adds hypothetical lifetime args to mentions of non-rewritten structs

### ISSUE THREAD

**Issue by spernsteiner** [immunant/c2rust#994]
```Rust
extern "C" {
    fn epoll_wait(events: *mut epoll_event);
}

pub struct fdevents {
    pub epoll_events: *mut epoll_event,
}

pub struct epoll_event {
    pub ptr: *mut u8,
}
```

Rewritten output:
```Rust
extern "C" {
    fn epoll_wait(events: *mut epoll_event);
}

pub struct fdevents<'h4,'h5> {
    pub epoll_events: &'h4 (epoll_event<'h5>),
}

pub struct epoll_event {
    pub ptr: *mut u8,
}
```

`struct epoll_event` can't be rewritten, since it's exposed to FFI via `fn epoll_wait`.  But the field type in `fdevents` is rewritten as if a lifetime parameter was added to `epoll_event`.

**Comment by kkysen** [comment]
`struct epoll_event`'s fields are marked `FIXED` as per #960, right?  This just needs to skip adding hypothetical lifetimes to `FIXED` fields/types, rights?

**Comment by spernsteiner** [comment]
Yes, something like that
